### PR TITLE
[Fix #49] Move `/app/mirage` to `/mirage`

### LINKED ISSUE #49: Move `/app/mirage` to `/mirage`
Is this possible?

### PR DESCRIPTION
**Pull request by dschmidt**
Hey,

this basically moves `/app/mirage` to `/mirage` while keeping the exact former behavior afaict:
- /mirage is watched and auto rebuilds the project if files are changed
- $addonDir/app is also merged into the app tree but not watched
- blueprints respect the new structure
- tests pass
- dummy app works too afaict

motivation:
sharing one mirage folder between multiple ember applications built in one project

BR
Domme

### CODE REVIEW

**Comment by dschmidt** [comment]
Ah, mocha node tests weren't run for me, will fix now.

**Comment by jherdman** [comment]
> sharing one mirage folder between multiple ember applications built in one project

Out of curiosity, how were you planning on doing this? My team has talked about this possibility a few times, and we had thrown around the idea of a shared add-on that is our Mirage config/factories/etc.

**Comment by dschmidt** [comment]
Well, the PR as implemented right now expects the mirage folder to be in `$project/mirage`, which is already good enough for me.

If you need more flexibility, you can also override it via
```javascript
EmberApp({
  mirage: {
    directory: wherever
  }
});
```
where `wherever` is a broccoli tree/node, so you could even do funky merging of different mirage trees before passing it to the addon.

If you want any more details on my setup, feel free to chat me up on Slack or IRC (freenode), in both cases my nick domme ;-)

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "active-model-adapter": "1.13.6",
     "broccoli-asset-rev": "^2.1.2",
     "broccoli-static-compiler": "^0.2.1",
+    "broccoli-unwatched-tree": "^0.1.1",
```
Comment (samselikoff) [inline package.json]: I'm getting an error using your fork on a new app, I think this needs to be in `dependencies`

Comment (dschmidt) [inline package.json]: Yeah, it totally needs to be.

**Comment by dschmidt** [comment]
I knew it, when I saw the other PR. Will rebase ;-)

**Comment by trabus** [comment]
This is awesome. I've been wanting this since I heard about mirage at EmberConf.

**Comment by samselikoff** [comment]
Ok I tried it out, this lgtm. Super exciting dude! Even in a dummy project it just *feels* right having it outside of `/app`. Pumped about this.

One nit to pick, could you change the config from `mirage` to `ember-cli-mirage`, just to be consistent with the `environment.js` option, then squash + I'll merge? Thank you SO MUCH for your work!

**Comment by dschmidt** [comment]
Hehe, you're welcome. As you know I'm scratching my own itch, happy I can land a breaking change so easily and didn't have to do endless hacks to make the change backwards compatible :)

Btw, I already squashed the commits, but if you want to review only the config option change, you can see it in isolation here: https://github.com/samselikoff/ember-cli-mirage/commit/cb62e7764b7e56ca2e313f5350d42dcf04a873c6

**Comment by samselikoff** [comment]
I trust your work. thanks again man!

**Pull request merged**